packages/intersection-observer/src/index.ts

```diff
@@ -57,31 +57,97 @@ export type E = JSX.Element;
   const stop = () => instance.disconnect();
+  const reset = () => instance.takeRecords().forEach((el) => remove(el.target));
   onMount(start);
   onCleanup(stop);
-  return [add, { remove, start, stop, instance }];
+  return { add, remove, start, stop, reset, instance };
+};
+
+/**
+ * Creates a reactive Intersection Observer primitive.
+ *
+ * @param elements - A list of elements to watch
+ * @param onChange - An event handler that returns an array of observer entires
+ * @param options - IntersectionObserver constructor options:
+ * - `root` — The Element or Document whose bounds are used as the bounding box when testing for intersection.
+ * - `rootMargin` — A string which specifies a set of offsets to add to the root's bounding_box when calculating intersections, effectively shrinking or growing the root for calculation purposes.
+ * - `threshold` — Either a single number or an array of numbers between 0.0 and 1.0, specifying a ratio of intersection area to total bounding box area for the observed target.
+ *
+ * @example
+ * ```tsx
+ * const createIntersectionObserver(els, entries =>
+ *   console.log(entries)
+ * );
+ * ```
+ */
+export const createIntersectionObserver = (
+  elements: MaybeAccessor<Element[]>,
+  onChange: IntersectionObserverCallback,
+  options?: IntersectionObserverInit
+) => {
+  const { add, reset } = makeIntersectionObserver([], onChange, options);
+  createEffect(() => {
+    reset();
+    access(elements).forEach(el => add(el));
+  });
+};
+
+/**
+ * Creates reactive signal that changes when a single element's visibility changes.
+ *
+ * @param element - An element to watch
+ * @param options - A Primitive and IntersectionObserver constructor options:
+ * - `root` — The Element or Document whose bounds are used as the bounding box when testing for intersection.
+ * - `rootMargin` — A string which specifies a set of offsets to add to the root's bounding_box when calculating intersections, effectively shrinking or growing the root for calculation purposes.
+ * - `threshold` — Either a single number or an array of numbers between 0.0 and 1.0, specifying a ratio of intersection area to total bounding box area for the observed target.
+ * - `initialValue` — Initial value of the signal *(default: false)*
+ * - `once` — If true: the stop function will be called automatically after visibility changes *(default: false)*
+ *
+ * @example
+ * ```ts
+ * let el!: HTMLElement
+ * const [isVisible, { start, stop, instance }] = createVisibilityObserver(() => el, { once: true })
+ * ```
+ */
+export const createVisibilityObserver = (
+  element: MaybeAccessor<Element>,
+  options?: IntersectionObserverInit & {
+    initialValue?: boolean;
+    once?: boolean;
+  }
+): [Accessor<boolean>, { start: () => void; stop: () => void; instance: IntersectionObserver }] => {
+  const [isVisible, setVisible] = createSignal(options?.initialValue ?? false);
+  const { start, stop, instance } = makeIntersectionObserver(
+    [access(element)],
```

Comment:
I believe this isn't reactive now, if the element in an accessor. It would have to be `createIntersectionObserver` for an accessor, and `makeIntersectionObserver` for just element.

Reply:
Oops good point. I added an effect if an accessor is supplied.